File: maven-plugin/src/main/java/org/contractfirst/generator/mavenplugin/MavenLogAdapter.java
package org.contractfirst.generator.mavenplugin;

import org.apache.maven.plugin.logging.Log;
import org.contractfirst.generator.logging.LogAdapter;

/**
 * Logging adapter for Maven Log.
 */
class MavenLogAdapter implements LogAdapter {

  private final Log underlying;

  MavenLogAdapter(Log underlying) {
    this.underlying = underlying;
  }

  @Override
  public boolean isDebugEnabled() {
    return underlying.isDebugEnabled();
  }

  @Override
  public void debug(String msg) {
    underlying.debug(msg);
  }

  @Override
  public void debug(String msg, Throwable error) {
    underlying.debug(msg, error);
  }

  @Override
  public boolean isInfoEnabled() {
    return underlying.isInfoEnabled();
  }

  @Override
  public void info(String msg) {
    underlying.info(msg);
  }

  @Override
  public void info(String msg, Throwable error) {
    underlying.info(msg, error);
  }

  @Override
  public boolean isWarnEnabled() {
    return underlying.isWarnEnabled();
  }

  @Override
  public void warn(String msg) {
    underlying.warn(msg);
  }

  @Override
  public void warn(String msg, Throwable error) {
    underlying.warn(msg, error);
  }

  @Override
  public boolean isErrorEnabled() {
    return underlying.isErrorEnabled();
  }

  @Override
  public void error(String msg) {
    underlying.error(msg);
  }

  @Override
  public void error(String msg, Throwable error) {
    underlying.error(msg, error);
  }
}

File: maven-plugin/src/main/kotlin/de/rk42/openapi/codegen/mavenplugin/MavenLogAdapter.kt
package de.rk42.openapi.codegen.mavenplugin

import de.rk42.openapi.codegen.logging.LogAdapter
import org.apache.maven.plugin.logging.Log

/**
 * Logging adapter for Maven Log.
 */
class MavenLogAdapter(private val underlying: Log) : LogAdapter {

  override fun isDebugEnabled(): Boolean {
    return underlying.isDebugEnabled
  }

  override fun debug(msg: String) {
    underlying.debug(msg)
  }

  override fun debug(msg: String, error: Throwable) {
    underlying.debug(msg, error)
  }

  override fun isInfoEnabled(): Boolean {
    return underlying.isInfoEnabled
  }

  override fun info(msg: String) {
    underlying.info(msg)
  }

  override fun info(msg: String, error: Throwable) {
    underlying.info(msg, error)
  }

  override fun isWarnEnabled(): Boolean {
    return underlying.isWarnEnabled
  }

  override fun warn(msg: String) {
    underlying.warn(msg)
  }

  override fun warn(msg: String, error: Throwable) {
    underlying.warn(msg, error)
  }

  override fun isErrorEnabled(): Boolean {
    return underlying.isErrorEnabled
  }

  override fun error(msg: String) {
    underlying.error(msg)
  }

  override fun error(msg: String, error: Throwable) {
    underlying.error(msg, error)
  }
}